ElectricityBillApi.Tests/ElectricProviderPickerServiceTests.cs

```diff
@@ -157,10 +134,13 @@ private IEnumerable<ElectricProvider> GetTestData()
             providerExpensiveNear.Subscribe(plantExpensive);
             providerExpensiveFar.Subscribe(plantExpensiveFar);
 
-            yield return providerCheapNear;
-            yield return providerCheapFar;
-            yield return providerExpensiveNear;
-            yield return providerExpensiveFar;
+            return new List<ElectricProvider>() 
+            {
+                providerCheapNear,
+                providerCheapFar,
+                providerExpensiveNear,
+                providerExpensiveFar
+            };
```

Comment:
Yield return was the right way to do this. The downside of doing what you did now is that you always have 4 providers in memory. `yield return` is lazy loading- you only ever have 1 provider at a time and when the next case runs- the previous provider will be forgotten.